rules/Arguments/NodeAnalyzer/ChangedArgumentsDetector.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Arguments\NodeAnalyzer;
+
+use PhpParser\Node\Param;
+use Rector\Core\PhpParser\Node\Value\ValueResolver;
+use Rector\NodeNameResolver\NodeNameResolver;
+
+final class ChangedArgumentsDetector
+{
+    public function __construct(
+        private ValueResolver $valueResolver,
+        private NodeNameResolver $nodeNameResolver
+    ) {
+    }
+
+    /**
+     * @param mixed $value
+     */
+    public function isDefaultValueChanged(Param $param, $value): bool
+    {
+        if ($param->default === null) {
+            return false;
+        }
+
+        return ! $this->valueResolver->isValue($param->default, $value);
+    }
+
+    public function isTypeChanged(Param $param, ?string $tyoe): bool
+    {
+        if ($param->type === null) {
+            return false;
+        }
+
+        if ($tyoe === null) {
```

Comment:
typo: `tyoe` => `type`?

```suggestion
        if ($type === null) {
```